---
title: "ChIP-seq Analysis of TGF-beta Treated SMAD2 and SMAD3 Peaks"
output:
  html_notebook:
    theme: flatly
    toc: true
    toc_float:
      smooth_scroll: true
      collapsed: false
    number_sections: true
---


# Introduction

 Smad2 and Smad3 are both DNA-binding proteins that are activated by TGFβ signalling, and have differing downstream targets and phenotypic effects. The predominance of either the Smad2 or the Smad3 pathway over the other can determine the overall effect of signalling TGFβ . The aim of this analysis was to determine the difference between the Smad2 and the Smad3 axes of the TGFβ pathway, by looking at already-called ChIP-seq peaks for Smad2 and Smad3, and comparing the functional significance of their corresponding genes by looking at associated Gene Ontology terms.

# Methods and Code

This section contains details of the analysis method and the code used to implement the analysis.

## Libraries

The following libraries were loaded during this analysis:

1. tidyverse

2. formatR

3. GenomicRanges

4. org.Hs.eg.db

5. TxDb.Hsapiens.UCSC.hg38.knownGene

6. ChIPseeker

7. clusterProfiler

8. kableExtra


```{r, echo=FALSE, include = FALSE}

library(tidyverse)
library(formatR)
library(GenomicRanges)
library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(ChIPseeker)
library(clusterProfiler)
library(kableExtra)
```


## Annotating Peaks

Peaks were annotated using the **ChIPseeker** package. The annotated peaks for each of the conditions were further filtered into three subsets:

1. All peaks 

2. Peaks with only promoter annotations. 

3. Peaks lying within genic regions, i.e., peaks without the annotations "Distal Intergenic" and "Downstream"

The following code was used to read in the file names and create a character vector from them. Those names were then fed into a function to convert those files into Granges objects held in a list. 

```{r, tidy=TRUE, echo = TRUE}
# 
# knitr::opts_chunk$set(echo = FALSE, tidy = TRUE)

## Takes a filename as string input, and if it has suffix ".narrowPeak", converts it to a GRanges object containing a q_value metadata column. Requires tidyverse and GRanges packages. 

## checking for existence of file and correct extension
narrow_to_granges <- function(file){stopifnot(file.exists(file), endsWith(x = file, suffix = ".narrowPeak")) 
  
  
  bed <- as_tibble(read_tsv(file = file, col_names = c("chrom", "start", "end",   "name", "score", "strand", "signal_value", "p_value", "q_value", "peak"))) 
  
  
  gr <- GRanges(seqnames = bed$chrom, ranges = IRanges(start = bed$start, end =   bed$end), strand = if_else(condition = bed$strand == "+"|  bed$strand == "-",   true = bed$strand, false = "*"), q_value = bed$q_value)
  
  return(gr)}


### To read in multiple .narrowPeak files and create a GRanges list of GRanges objects with the input of respective files.   

files <- dir(path = ".", pattern = "\\.narrowPeak$") ## specify files in character vector

grl <- lapply(files, narrow_to_granges) ## Create list of GRanges object

```


Here are the contents of the files vector. 

```{r, eval=TRUE, tidy=TRUE, echo=TRUE}
files
```

The following code was then used to assign the Granges in the list to separate variables. The name of the variable to hold each GRange object was derived from the respective file name from which its data was obtained. The names of the variables were stored as a character vector "v".To names in v, the respective GRange was assigned.  


```{r, tidy=TRUE, echo=TRUE}

TxDb <- TxDb.Hsapiens.UCSC.hg38.knownGene

## vector of variable names for granges in list

v <- str_remove_all(string = files, pattern = "\\.narrowPeak$") 

### Assign individual granges within grl to variable names in v. 

for(i in seq_along(grl)){
  assign(x = v[i], value = grl[[i]])
}


```


Here are the names of the variables held in vector v:

```{r, eval=TRUE, tidy=TRUE, echo=TRUE}
v
```


A character vector of variable names was created to hold the annotated peak objects derived from the GRanges. 



```{r tidy=TRUE, echo=TRUE}
ap_v <- paste0("AP_", v) ## variable names for annotated peak objects

ap_v
### Create annotated peak objects and assign them to names in ap_v
```

```{r tidy=TRUE, echo=TRUE}
for(i in seq_along(grl)){
  assign(x = ap_v[i], value = annotatePeak(peak = grl[[i]], TxDb = TxDb, annoDb = "org.Hs.eg.db"))
}
```


```{r tidy=TRUE, echo=TRUE}
AP_SMAD2_abInput_treated_peaks
```


```{r tidy=TRUE, echo=TRUE}
df_ap_v <- paste0("df_", ap_v) ## variable names for tibbles derived from annotated peaks

df_ap_v
```


```{r tidy=TRUE, echo=TRUE}
### Create tibbles from annotated peak objects listed in ap_v and assign them to names in df_ap_v

for(i in seq_along(ap_v)){
  assign(x = df_ap_v[i], value = parse(text = ap_v[i]) %>% 
           eval() %>% 
           AnnotationDbi::as.data.frame() %>% 
           as_tibble()) 
}
```


```{r  tidy=TRUE, echo=TRUE}
df_AP_SMAD2_abInput_treated_peaks

```

```{r tidy=TRUE, echo=TRUE}
## variable names for tibbles derived from annotated peaks limited to promoter annotations

prom_df_ap_v <- paste0("prom_df_", ap_v)
prom_df_ap_v
```


```{r tidy=TRUE, echo=TRUE}
### Create subset of tibbles from variable names listed in df_ap_v and assign them to variable names in prom_df_ap_v
for(i in seq_along(df_ap_v)){
  assign(x = prom_df_ap_v[i], value = parse(text = df_ap_v[i]) %>% 
           eval() %>% 
           filter(str_detect(annotation, "Promoter")))
}

prom_df_AP_SMAD2_abInput_treated_peaks

```



```{r tidy=TRUE, echo=TRUE}
## variable names for tibbles derived from annotated peaks limited to gene region annotations

gen_df_ap_v <- paste0("gen_df_", ap_v) 

gen_df_ap_v
```


```{r tidy=TRUE, echo=TRUE}
### Filter tibbles with variable names listed in df_ap_v to discard rows with "Distal Intergenic" or "Downstream" annotations and assign them to names in gen_df_ap_v. 

for(i in seq_along(df_ap_v)){
  assign(x = gen_df_ap_v[i], value = parse(text = df_ap_v[i]) %>% 
           eval() %>% 
           filter(!str_detect(annotation, "Distal Intergenic|Downstream")))
}
```


```{r tidy=TRUE, echo=TRUE}
gen_df_AP_SMAD3_LAP_treated_peaks
```

## GO Term Enrichment Analysis of Annotated Peaks

GO term enrichment analysis of the annotated peaks was carried out using the **ClusterProfiler** package. Analysis was carried out for all four conditions and each subset of annotated peaks  within each condition (all peaks, promoter-limited peaks and genic region-limited peaks). 

**Note:** The following code takes *very* long to run.

### GO term enrichment for all peaks


```{r echo = FALSE}
load(file = "annotation_go_results.RData")
```



```{r eval = FALSE, tidy = TRUE, echo=TRUE}

ego_v <- paste0("ego_", v) ### Vector of variable names for go term enrichment objects


for(i in seq_along(ego_v)){ 
  ego <- parse(text = df_ap_v[i]) %>% 
    eval(expr = .) %>% 
    dplyr::select(geneId) %>% 
    unlist() %>% 
    enrichGO(gene = ., OrgDb = org.Hs.eg.db, ont = "BP", pAdjustMethod = "BH", readable = TRUE)
  if(!nrow(as.data.frame(ego)) == 0) 
    assign(x = ego_v[i], value = ego)}


```


```{r tidy = TRUE, echo=TRUE}
ego_SMAD2_abInput_treated_peaks ## representative example of an enrichGO object
```

### GO term enrichment for promoter-proximal peaks

```{r tidy = TRUE, echo=TRUE}
### Vector of variable names for go term enrichment objects limited to promoter peaks

prom_ego_v <- ego_v <- paste0("prom_ego_", v) 

prom_ego_v

```

```{r eval = FALSE, tidy = TRUE, echo=TRUE}
### Performing go term analysis using the clusterprofiler package and assigning the results to the variables in prom_ego_v

for(i in seq_along(prom_ego_v)){ 
  ego <- parse(text = prom_df_ap_v[i]) %>% 
    eval(expr = .) %>% 
    dplyr::select(geneId) %>% 
    unlist() %>% 
    enrichGO(gene = ., OrgDb = org.Hs.eg.db, ont = "BP", pAdjustMethod = "BH", readable = TRUE)
  if(!nrow(as.data.frame(ego)) == 0) 
    assign(x = prom_ego_v[i], value = ego)}
```


```{r tidy = TRUE, echo=TRUE}
prom_ego_SMAD2_abInput_treated_peaks

```

### GO term enrichment for peaks lying within genic regions

```{r tidy = TRUE, echo=TRUE}
### Vector of variable names for go term enrichment objects limited to genic region peaks

gen_ego_v <- paste0("gen_ego_", v)

gen_ego_v
```


```{r eval=FALSE, tidy = TRUE, echo=TRUE}
### Performing go term analysis using the clusterprofiler package and assigning the results to the variables in gen_ego_v

for(i in seq_along(gen_ego_v)){ 
    ego <- parse(text = gen_df_ap_v[i]) %>% 
      eval(expr = .) %>% 
      dplyr::select(geneId) %>% 
      unlist() %>% 
      enrichGO(gene = ., OrgDb = org.Hs.eg.db, ont = "BP", pAdjustMethod = "BH", readable = TRUE)
  if(!nrow(as.data.frame(ego))==0) 
    assign(x = gen_ego_v[i], value = ego)}
```


```{r tidy = TRUE, echo=TRUE}
rm(ego) ## removing unnecessary variable ego
```



```{r echo = FALSE}
save.image(file = "annotation_go_results.RData")
```



# Results

## Peak annotation results

### SMAD2 peaks - untreated

Most of the untreated SMAD2 peaks were in the distal intergenic regions (about 86\%).  

```{r tidy = TRUE}
plotAnnoPie(AP_SMAD2_abInput_untreated_peaks)
title(main = "SMAD2 Peaks: Untreated", line = -2, adj = 0)
```



### SMAD2 peaks - treated

Similar to the untreated SMAD2 peaks, most of the treated SMAD2 peaks also lay within the intergenic regions ( about 71\%), but in a lower proportion as compared to the untreated peaks (about 86\%). On the other hand, the proportion of SMAD2 peaks in the genic regions rose upon treatment from `r 100 - (0.35 + 86.37)`\% to `r 100 - (0.46 + 70.01)`\%. 

```{r tidy = TRUE}
plotAnnoPie(AP_SMAD2_abInput_treated_peaks)
title(main = "SMAD2 Peaks: Treated", line = -2, adj = 0)
```

### SMAD3 peaks - untreated

Around half of the untreated SMAD3 peaks were in the intergenic regions, while a majority of the peaks in the genic regions (`r 10.04 + 23.8` \%) overlapped with intronic sequences. Compared to the SMAD2 peaks (both treated and untreated), the SMAD3 untreated peaks had a far lower proportion of peaks within the intergenic regions. On the other hand, the proportion of SMAD3 peaks in intronic regions was noticeably elevated as compared to the SMAD2 peaks.


```{r tidy = TRUE}
plotAnnoPie(AP_SMAD3_LAP_untreated_peaks)
title(main = "SMAD3 Peaks: Untreated", line = -2, adj = 0)

```

### SMAD3 peaks - treated

TGF-$\beta$ ligand treatment reduced the proportion of SMAD3 peaks in the intergenic regions (`r 51.45 + 0.83`\% without treatment vs. `r 40.78 + 0.88`\% with treatment). For the most part, those peaks seem to have been displaced to the promoter regions, which accounted for a total of `r 5.74 + 3.05 + 2.7`\% before treatment but rose to `r 9.71 + 4.61 + 3.71`\% after treatment.     


```{r tidy=TRUE}
plotAnnoPie(AP_SMAD3_LAP_treated_peaks)
title(main = "SMAD3 Peaks: Treated", line = -2, adj = 0)
```


## GO term analysis results

GO term analysis was limited to biological process (BP) terms; GO Terms relating to Cellular Component and Molecular Function were not analysed.


### GO term analysis for all peaks

No enriched BP GO terms were found within annotations for untreated SMAD2 peaks. This may have been due to the majority of these peaks being located in the intergenic regions. For treated SMAD2 peaks, GO terms related to processes in neuronal development were over-represented. 

**Note:** Only the first 20 enriched GO terms for each peak subset and condition are shown. Terms were arranged in ascending order by adjusted p value, then in descending order by gene ratio and absolute count. 


```{r}
ego_SMAD2_abInput_treated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count), desc(GeneRatio)) %>% filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Treated SMAD2 Peaks")
```

```{r tidy = TRUE}
dotplot(ego_SMAD2_abInput_treated_peaks) + 
  ggtitle(label = "Enriched BP GO Terms in Treated SMAD2 Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))

```

Untreated SMAD3 peaks also showed an enrichment in neuronal development-related terms. In addition, peaks in regions related to GTPase activity and cell-cell adhesion were also enriched.  

```{r}
ego_SMAD3_LAP_untreated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Untreated SMAD3 Peaks")
```

```{r}
dotplot(ego_SMAD3_LAP_untreated_peaks) + 
  ggtitle(label = "Enriched BP GO Terms in Untreated SMAD3 Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))
```

Similar terms related to neuronal development also appeared in treated SMAD3 peaks. However, additional enriched terms such as "regulation of small GTPase-mediated signal transduction", as well as "regulation of Ras protein signal transduction" were also found to be enriched. 

```{r}

ego_SMAD3_LAP_treated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Treated SMAD3 Peaks")
```

```{r}
dotplot(ego_SMAD3_LAP_treated_peaks) + 
  ggtitle(label = "Enriched BP GO Terms in Treated SMAD3 Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))

```

### GO Term Analysis for Peaks Restricted to Gene Regions

The enriched GO Terms for SMAD2 peaks limited to gene regions did not differ much from their counterparts where all the peaks were considered. The pattern remains the same, with terms associated with neuronal development being predominantly featured, although terms for limb and appendage morphogenesis also turn up. 


```{r}
gen_ego_SMAD2_abInput_treated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio)) %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Treated SMAD2 Peaks Restricted to Gene Regions")
```

```{r}
dotplot(gen_ego_SMAD2_abInput_treated_peaks) + ggtitle(label = "Enriched BP GO Terms in Treated SMAD2 Genic Peaks") + theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))
```

Untreated SMAD3 peaks in gene regions showed a similar neuronal development theme for the most significantly enriched genes. 

```{r}
gen_ego_SMAD3_LAP_untreated_peaks@result %>% 
  as_tibble() %>% arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Untreated SMAD3 Peaks Restricted to Gene Regions")
```

```{r}
dotplot(gen_ego_SMAD3_LAP_untreated_peaks ) + 
  ggtitle(label = "Enriched BP GO Terms in Untreated SMAD3 Genic Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))
```

Treated SMAD3 peaks in gene regions did not show too many chages either, as compared to the the set of all treated SMAD3 peaks.

```{r}
gen_ego_SMAD3_LAP_treated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% filter(p.adjust < 0.05) %>%  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Treated SMAD3 Peaks Restricted to Gene Regions")
```

```{r}
dotplot(gen_ego_SMAD3_LAP_treated_peaks ) + 
  ggtitle(label = "Enriched BP GO Terms in Treated SMAD3 Genic Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))

```

### GO Term Analysis for Peaks Restricted to Promoter Regions

When peaks were limited to promoter regions, the enriched GO term sets showed noticeable differences from sets derived from all peaks as well as sets derived from peaks limited to gene regions. No enriched GO terms were found for treated promoter-proximal SMAD2 peaks. 


```{r}
prom_ego_SMAD2_abInput_untreated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Untreated SMAD2 Peaks Restricted to Promoters")
```

```{r}
dotplot(prom_ego_SMAD2_abInput_untreated_peaks) + ggtitle(label = "Enriched BP GO Terms in Untreated SMAD2 Promoter Peaks") + theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))
```

Untreated SMAD3 peaks restricted to promoters showed the least number of enriched terms, all of which were related to cell-cell adhesion. 

```{r}
prom_ego_SMAD3_LAP_untreated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>% 
  rename(Description = "Enriched GO Terms in Untreated SMAD3 Peaks Restricted to Promoters")
```


```{r}
dotplot(prom_ego_SMAD3_LAP_untreated_peaks) + 
  ggtitle(label = "Enriched BP GO Terms in Untreated SMAD3 Promoter Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 5))
```

Treatment with ligand seemed to show many more enriched GO terms in promoter-restricted SMAD3 peaks, as compared to untreated promoter-restricted SMAD3 peaks. 

```{r}
prom_ego_SMAD3_LAP_treated_peaks@result %>% 
  as_tibble() %>% 
  arrange(p.adjust, desc(Count) , desc(GeneRatio))  %>% 
  filter(p.adjust < 0.05) %>%  
  dplyr::select(Description, p.adjust, GeneRatio, BgRatio) %>% 
  head(20) %>%
  rename(Description = "Enriched GO Terms in Treated SMAD3 Peaks Restricted to Promoters")
```

```{r}
dotplot(prom_ego_SMAD3_LAP_treated_peaks) + 
  ggtitle(label = "Enriched BP GO Term in Treated SMAD3 Promoter Peaks") + 
  theme(plot.title = element_text(hjust = 0.8, size = 15), axis.text.y = element_text(size = 7))
```

